sdk/core/core-http/lib/util/atomXmlHelper.ts

```diff
@@ -0,0 +1,338 @@
+  if (!atomResponseInJson) {
+    return;
+  }
+
+  if (atomResponseInJson.feed) {
+    return parseFeedResult(atomResponseInJson.feed);
+  }
+
+  if (atomResponseInJson.entry) {
+    return parseEntryResult(atomResponseInJson.entry);
+  }
+
+  throw new Error("Unrecognized result: " + JSON.stringify(atomResponseInJson));
+}
+
+/**
+ * @ignore
+ * Utility to help parse given `entry` result
+ * @param entry
+ */
+function parseEntryResult(entry: any): XMLResponseInJSON | undefined {
+  let result: XMLResponseInJSON;
+
+  if (
+    typeof entry !== "object" ||
+    entry == null ||
+    typeof entry.content !== "object" ||
+    entry.content == null
+  ) {
+    return undefined;
+  }
+
+  const contentElementNames = Object.keys(entry.content).filter(function(key) {
+    return key !== Constants.XML_METADATA_MARKER;
+  });
+
+  if (contentElementNames && contentElementNames[0]) {
+    const contentRootElementName = contentElementNames[0];
+    delete entry.content[contentRootElementName][Constants.XML_METADATA_MARKER];
+    result = entry.content[contentRootElementName];
+
+    if (result) {
+      if (entry[Constants.XML_METADATA_MARKER]) {
+        result[Constants.ATOM_METADATA_MARKER] = entry[Constants.XML_METADATA_MARKER];
+      } else {
+        result[Constants.ATOM_METADATA_MARKER] = {};
+      }
+
+      result[Constants.ATOM_METADATA_MARKER]["ContentRootElement"] = contentRootElementName;
+
+      Object.keys(entry).forEach((property: string) => {
+        if (property !== "content" && property !== Constants.XML_METADATA_MARKER) {
+          result[Constants.ATOM_METADATA_MARKER][property] = entry[property];
+        }
+      });
+
+      return result;
+    }
+  }
+
+  return undefined;
+}
+
+/**
+ * @ignore
+ * Utility to help parse given `feed` result
+ * @param feed
+ */
+function parseFeedResult(feed: any): XMLResponseInJSON[] {
+  const result = [];
+  if (typeof feed === "object" && feed != null && feed.entry) {
+    if (Array.isArray(feed.entry)) {
+      feed.entry.forEach((entry: any) => {
+        const parsedEntryResult = parseEntryResult(entry);
+        if (parsedEntryResult) {
+          result.push(parsedEntryResult);
+        }
+      });
+    } else {
+      const parsedEntryResult = parseEntryResult(feed.entry);
+      if (parsedEntryResult) {
+        result.push(parsedEntryResult);
+      }
+    }
+  }
+  return result;
+}
+
+/**
+ * Transforms response to contain the parsed data.
+ * @param nameProperties The set of 'name' properties to be constructed on the
+ * resultant object e.g., QueueName, TopicName, etc.
+ * @param response
+ * @param shouldParseResponse
+ */
+export async function deserializeAtomXmlResponse(
+  nameProperties: string[],
+  response: HttpOperationResponse,
+  shouldParseResponse: boolean
+): Promise<HttpOperationResponse> {
+  let atomResponseInJson: any;
+  try {
+    if (response.bodyAsText && response.bodyAsText.toString().length > 0) {
```

Comment:
Is there any scenario where bodyAsText is not a string? I understand it could be null or undefined, but can it be anything else? If not, then the above if condition can be reduced to `if (response.bodyAsText)`

Reply:
This is so we can do the check - length > 0

Reply:
Assuming `bodyAsText` is a string, `if (response.bodyAsText && response.bodyAsText.length > 0)` and `if (response.bodyAsText)` would be the same check.
This is because `if (response.bodyAsText)` check fails for empty string.

Even if you want to keep the length > 0 check explicitly, we still shouldn't need the `.toString()`